src/AppInstallerRepositoryCore/Microsoft/PredefinedInstalledSourceFactory.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "pch.h"
+#include "Microsoft/PredefinedInstalledSourceFactory.h"
+#include "Microsoft/SQLiteIndex.h"
+#include "Microsoft/SQLiteIndexSource.h"
+
+using namespace std::string_literals;
+using namespace std::string_view_literals;
+
+namespace AppInstaller::Repository::Microsoft
+{
+    namespace
+    {
+        // Populates the index with the ARP entries from the given root.
+        void PopulateIndexFromARP(SQLiteIndex& index, HKEY rootKey)
+        {
+            UNREFERENCED_PARAMETER(index);
+            UNREFERENCED_PARAMETER(rootKey);
+        }
+
+        // Populates the index with the entries from MSIX.
+        void PopulateIndexFromMSIX(SQLiteIndex& index)
+        {
+            using namespace winrt::Windows::ApplicationModel;
+            using namespace winrt::Windows::Management::Deployment;
+
+            PackageManager packageManager;
+            auto packages = packageManager.FindPackagesForUserWithPackageTypes({}, PackageTypes::Main);

```

Comment:
>PackageTypes::Main [](start = 83, length = 18)

Do we need a TODO for other types?

Reply:
What other types would you expect to enumerate?
    enum class PackageTypes : uint32_t
    {
        None = 0,
        Main = 0x1,
        Framework = 0x2,
        Resource = 0x4,
        Bundle = 0x8,
        Xap = 0x10,
        Optional = 0x20,
    };

---
In reply to: [507496922](https://github.com/microsoft/winget-cli/pull/598#discussion_r507496922) [](ancestors = 507496922)

Reply:
I guess we might enable optional once we have dependencies working?

---
In reply to: [508136073](https://github.com/microsoft/winget-cli/pull/598#discussion_r508136073) [](ancestors = 508136073,507496922)

Reply:
Added a TODO to consider optional packages

---
In reply to: [508228016](https://github.com/microsoft/winget-cli/pull/598#discussion_r508228016) [](ancestors = 508228016,508136073,507496922)